### PR TITLE
🔖 Release 3.5.2

### PR DESCRIPTION
**Pull request by ayushuk**
* #351
* #346
* #345

### CODE REVIEW

File: pros/cli/conductor.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def upgrade(ctx: click.Context, project: c.Project, query: c.BaseTemplate, **kwa
     """
     analytics.send("upgrade-project")
     if not query.name:
-        for template in project.templates.keys():
+        for template in tuple(project.templates.keys()):
```
Comment (noam987) [inline pros/cli/conductor.py:162]: Just to be sure, this won't break if they have old conductor files?

**Review by noam987 — COMMENTED** [review]
One comment. if it is tested and won't be an issue, LGTM and I will apprrove

File: pros/cli/conductor.py (hunk 1 above)
Comment (ayushuk) [reply pros/cli/conductor.py:162]: From my understanding, this function doesn't touch the conductor file (there are no `.save()` calls).

**Review by noam987 — APPROVED** [review]
LGTM

**Pull request merged**